### PR TITLE
Add new tests and features to Solana ragger

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

**Comment by fbeutin-ledger** [comment]
SonarCloud is analyzing the python tests, which is not what we want

File: .github/workflows/ci-workflow.yml
Comment (agrojean-ledger) [inline .github/workflows/ci-workflow.yml:10]: C'est mieux que ce qu'il y avait avant ?

Comment (fbeutin-ledger) [reply .github/workflows/ci-workflow.yml:10]: The goal is to trigger the CI even if the target branch of the PR is not develop
My opinion is that if it's a PR it needs a CI
It's useful for me as I opened two newer PRs that cascade into each other

File: tests/python/apps/solana.py
Comment (askibin) [inline tests/python/apps/solana.py:98]: first batch shouldn't have P2_EXTEND bit set. otherwise the app would try to extend the data buffer with the new data instead of resetting context

Comment (fbeutin-ledger) [reply tests/python/apps/solana.py:98]: Nice catch, I didn't test a lot the case where the message has to be split and sent in batches

**Comment by askibin** [comment]
does this need more work or can be merged?

**Comment by fbeutin-ledger** [comment]
I'm waiting for a final review from Ledger colleagues and then we'll merge the first two PRs

**Review by lpascal-ledger — APPROVED** [review]

**Pull request merged**
